DigitalLearningSolutions.Data/Services/SupervisorDelegateService.cs

```diff
@@ -30,17 +34,27 @@ ISupervisorDelegateDataService supervisorDelegateDataService
             return supervisorDelegateDataService.GetSupervisorDelegateRecordByInviteHash(inviteHash);
         }
 
-        public IEnumerable<SupervisorDelegate> GetPendingSupervisorDelegateRecordsByEmailAndCentre(
-            int centreId,
-            string email
-        )
+        public void AddDelegateIdToSupervisorDelegateRecords(IEnumerable<int> supervisorDelegateIds, int delegateId)
         {
-            return supervisorDelegateDataService.GetPendingSupervisorDelegateRecordsByEmailAndCentre(centreId, email);
+            supervisorDelegateDataService.UpdateSupervisorDelegateRecordsCandidateId(supervisorDelegateIds, delegateId);
         }
 
-        public void AddDelegateIdToSupervisorDelegateRecords(IEnumerable<int> supervisorDelegateIds, int delegateId)
+        public IEnumerable<SupervisorDelegate> GetPendingSupervisorDelegateRecordsByEmailsAndCentre(
+            int centreId,
+            IEnumerable<string?> emails
+        )
         {
-            supervisorDelegateDataService.UpdateSupervisorDelegateRecordsCandidateId(supervisorDelegateIds, delegateId);
+            emails = emails.Where(e => !string.IsNullOrWhiteSpace(e)).ToList();

```

Comment:
It's normally not considered good practice to mutate input parameters to a function, so I think I'd prefer this to be 
```
var nonEmptyEmails = emails.Where(e => !string.IsNullOrWhiteSpace(e)).ToList();
```